controllers/machinehealthcheck_controller.go

```diff
@@ -432,17 +453,19 @@ func (r *MachineHealthCheckReconciler) indexMachineByNodeName(object runtime.Obj
 // isAllowedRemediation checks the value of the MaxUnhealthy field to determine
 // whether remediation should be allowed or not
 func isAllowedRemediation(mhc *clusterv1.MachineHealthCheck) bool {
+	// TODO(JoelSpeed): return an error from isAllowedRemediation when maxUnhealthy
+	// is nil, we expect it to be defaulted always.
 	if mhc.Spec.MaxUnhealthy == nil {
 		return true
 	}
-	maxUnhealthy, err := intstr.GetValueFromIntOrPercent(mhc.Spec.MaxUnhealthy, int(mhc.Status.ExpectedMachines), false)
+
+	maxUnhealthy, err := getMaxUnhealthy(mhc)
 	if err != nil {
 		return false
 	}
 
 	// If unhealthy is above maxUnhealthy, short circuit any further remediation
```

Comment:
This may be a nit, but I'd be very clear with this comment. I say this because the phrase "short circuit" may require additional "thinking" especially for those who are not fluent in English or Electronics 😄 .
```suggestion
	// Remediation is not allowed if unhealthy is above maxUnhealthy
```